app/forms/questionnaire_form.py

```diff
@@ -134,18 +202,23 @@ def generate_form(schema, block_json, data, answer_store):
 
     class DynamicForm(QuestionnaireForm):
         date_ranges = []
+        calculated_question = []
 
     for question in schema.get_questions_for_block(block_json):
         if question['type'] == 'DateRange':
             DynamicForm.date_ranges += [(question['id'], question['answers'][0]['id'], question['answers'][1]['id'])]
+            DynamicForm.date_ranges += [(question['id'], question['answers'][0]['id'], question['answers'][1]['id'])]
+        elif question['type'] == 'Calculated':
+            target_answer = schema.get_answer(question['calculated']['answer_id'])
+            DynamicForm.calculated_question += [(question, target_answer)]
```

Comment:
I thought we were getting rid of all this logic here and moving it to `validate(self)`

Reply:
It breaks the dynamic form stuff, theres a card in our backlog to fix, so for now staying here

Reply:
We don't need to get rid of `DynamicForm` but we can get rid of the variables as we can work them out on validation

Reply:
Yep, having tried getting rid of `DynamicForm` here isn't as easy as I expected. However removing the `date_ranges` and `calculated_question` objects took 15 minutes.